pennylane_qiskit/converter.py

```diff
@@ -474,3 +560,31 @@ def load_qasm_from_file(file: str):
         function: the new PennyLane template
     """
     return load(QuantumCircuit.from_qasm_file(file))
+
+
+# pylint:disable=fixme, protected-access
+def _conditional_funcs(ops, cargs, operation_class, branch_funcs, ctrl_flow_type):
+    """Builds the conditional functions for Controlled flows"""
+    true_fn, false_fn, elif_fns = operation_class, None, ()
+    # Logic for using legacy c_if
+    if not isinstance(ops, ControlFlowOp):
+        return true_fn, false_fn, elif_fns, ops.condition
+
+    # Logic for handling IfElseOp
+    if ctrl_flow_type == "IfElseOp":
+        true_fn = branch_funcs[0]
+        if len(branch_funcs) == 2:
+            false_fn = branch_funcs[1]
+
+    # Logic for handling SwitchCaseOp
+    elif ctrl_flow_type == "SwitchCaseOp":
+        elif_fns = []
+        for res_bit, case in ops._case_map.items():
+            if not isinstance(case, _DefaultCaseType):
+                elif_fns.append((res_bit, branch_funcs[case]))
+        ops.condition = [tuple(cargs), "SwitchCase"]
+        if any((isinstance(case, _DefaultCaseType) for case in ops._case_map)):
+            true_fn = branch_funcs[-1]
+            ops.condition = [tuple(cargs), "SwitchDefault"]
```

Comment:
Why modify the `condition` attribute on `ops` instead of just unpacking the `ops.condition in the return in line 571 and having this function return `res_reg` and `res_big` instead of `ops.condition`. Do we use `ops.condition` later?

Reply:
This allows for a standardized way of storing the info we require since `SwitchOp` by default is `None`, so we are not overriding any information. I don't exactly use the info later in this PR but do so in the `expr` one, so would like to keep it.